### PR TITLE
bump argument parser version to 1.2.x

### PR DESCRIPTION
**Pull request by tomerd**

### CODE REVIEW

**Comment by tomerd** [comment]
https://github.com/apple/swift/pull/63521
https://github.com/apple/swift-driver/pull/1285
https://github.com/apple/sourcekit-lsp/pull/708

@swift-ci smoke test

**Comment by neonichu** [comment]
```
--- bootstrap: note: Building SwiftPM (with a freshly built swift-bootstrap)
Error: Unknown option '--build-path'
```

🤔

**Comment by MaxDesiatov** [comment]
That issue is fixed in https://github.com/apple/swift-argument-parser/pull/548, we just need a new version of SAP tagged.

**Comment by tomerd** [comment]
@neonichu maybe related to https://github.com/apple/swift/pull/63521#issuecomment-1423097899

**Comment by tomerd** [comment]
https://github.com/apple/swift/pull/63521
https://github.com/apple/swift-driver/pull/1285
https://github.com/apple/sourcekit-lsp/pull/708

@swift-ci smoke test

File: Package.swift [hunk 1]
```diff
@@ -692,7 +692,7 @@ if ProcessInfo.processInfo.environment["SWIFTCI_USE_LOCAL_DEPS"] == nil {
         // The 'swift-argument-parser' version declared here must match that
         // used by 'swift-driver' and 'sourcekit-lsp'. Please coordinate
         // dependency version changes here with those projects.
-        .package(url: "https://github.com/apple/swift-argument-parser.git", .upToNextMinor(from: "1.1.4")),
+        .package(url: "https://github.com/apple/swift-argument-parser.git", .upToNextMinor(from: "1.2.0")),
```
Comment (MaxDesiatov) [inline Package.swift:695]: It's probably preferable to fix that version without the regression as the lowest we allow, but also we could allow 1.3.0 and higher, assuming SAP maintains SemVer compatibility.
```suggestion
        .package(url: "https://github.com/apple/swift-argument-parser.git", from: "1.2.2"),
```

File: Package.swift [hunk 2]
```diff
@@ -692,7 +692,7 @@ if ProcessInfo.processInfo.environment["SWIFTCI_USE_LOCAL_DEPS"] == nil {
         // The 'swift-argument-parser' version declared here must match that
         // used by 'swift-driver' and 'sourcekit-lsp'. Please coordinate
         // dependency version changes here with those projects.
-        .package(url: "https://github.com/apple/swift-argument-parser.git", .upToNextMinor(from: "1.1.4")),
+        .package(url: "https://github.com/apple/swift-argument-parser.git", from: "1.2.2"),
```
Comment (tomerd) [inline Package.swift:695]:
```suggestion
        .package(url: "https://github.com/apple/swift-argument-parser.git", .upToNextMinor(from: "1.2.2")),
```

**Comment by tomerd** [comment]
https://github.com/apple/swift/pull/63521
https://github.com/apple/swift-driver/pull/1285
https://github.com/apple/sourcekit-lsp/pull/708

@swift-ci smoke test

**Comment by tomerd** [comment]
self hosted expected to fail since they do not support cross repo test

**Review by neonichu — APPROVED** [review]

**Review by MaxDesiatov — APPROVED** [review]

**Pull request merged**